protocol-units/bridge/service/src/types.rs

```diff
@@ -136,59 +136,13 @@ impl From<Uint<256, 4>> for TimeLock {
 }
 
 #[derive(Deref, DerefMut, Debug, Clone, Copy, PartialEq, Eq, Deserialize)]
-pub struct Amount(pub AssetType);
-
-impl Amount {
-	// pub fn value(&self) -> u64 {
-	// 	match self.0 {
-	// 		AssetType::EthAndWeth((weth_value, eth_value)) => weth_value + eth_value,
-	// 		AssetType::Moveth(value) => value,
-	// 	}
-	// }
-
-	pub fn moveth_value(&self) -> u64 {
-		match self.0 {
-			AssetType::EthAndWeth((_, _)) => 0,
-			AssetType::Moveth(value) => value,
-		}
-	}
-
-	pub fn eth_value(&self) -> u64 {
-		match self.0 {
-			AssetType::EthAndWeth((eth_value, _)) => eth_value,
-			AssetType::Moveth(_) => 0,
-		}
-	}
-
-	pub fn weth_value(&self) -> u64 {
-		match self.0 {
-			AssetType::EthAndWeth((_, weth_value)) => weth_value,
-			AssetType::Moveth(_) => 0,
-		}
-	}
-}
+pub struct Amount(pub u64);
```

Comment:
You use u64. If it's wei for example, we should use u128 because u64 is to small for important eth amount.

Reply:
`Uint<256, 4>` has support for conversion to u64 via its limbs methods but not for u128. 

This is a different version but the `Uint` methods are the same https://docs.rs/ruint/1.12.3/src/ruint/lib.rs.html#146